src/components/searchBar.tsx

```diff
@@ -0,0 +1,97 @@
+'use client'
+import {
+    Form,
+    FormControl,
+    FormDescription,
+    FormField,
+    FormItem,
+} from '@/components/ui/form'
+import { Input } from '@/components/ui/input'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { Loader2, Search } from 'lucide-react'
+import { useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import toast from 'react-hot-toast'
+import { z } from 'zod'
+import { Button } from './ui/button'
+import { useRouter } from 'next/navigation'
+
+const searchSchema = z.object({
+    searchTerm: z.string().min(1),
+})
+type searchSchema = z.infer<typeof searchSchema>
+export default function SearchBar({ prevSearch }: { prevSearch?: string }) {
+    const [isClicked, setIsClicked] = useState(false)
+    const router = useRouter()
+    const form = useForm<searchSchema>({
+        resolver: zodResolver(searchSchema),
+        defaultValues: {
+            searchTerm: prevSearch ?? '',
+        },
+    })
+    const [searchTerm, setSearchTerm] = useState('')
+
+    const submit = async (value: searchSchema) => {
+        try {
+            setIsClicked(true)
+            router.push(`/sok?searchTerm=${value.searchTerm}`)
+            setIsClicked(false)
+        } catch (error) {
+            toast.error('något gick fel')
+        }
+    }
+
+    // start of autocomplete function 
+    const asd = (value: string) => {
```

Comment:
I know it's just a placeholder but just having random `asd` function makes me think it's some other error and not pending progress, I would just have the onChange empty at line 54